src/mono/mono/mini/intrinsics.c

```diff
@@ -1007,6 +1007,54 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
 			EMIT_NEW_BIALU_IMM (cfg, ins, OP_COMPARE_IMM, -1, dreg, 0);
 			EMIT_NEW_UNALU (cfg, ins, OP_ICGT, dreg, -1);
 			ins->type = STACK_I4;
+			return ins;
+		} else if (!strcmp (cmethod->name, "CreateSpan") && fsig->param_count == 1) {
+			MonoGenericContext* ctx = mono_method_get_context (cmethod);
+			g_assert (ctx);
+			g_assert (ctx->method_inst);
+			g_assert (ctx->method_inst->type_argc == 1);
+			MonoType* arg_type = ctx->method_inst->type_argv [0];
+			MonoType* t = mini_get_underlying_type (arg_type);
+			g_assert (!MONO_TYPE_IS_REFERENCE (t) && t->type != MONO_TYPE_VALUETYPE);
+
+			// The following relies on the previous instruction being an OP_MOVE. Specifically. one that
+			// is emitted from CEE_TDTOKEN as the last EMIT_NEW_TEMPLOAD, which has its inst_p1 set.
+			// Bail out and use the non-intrinsic variant if this is not satisfied.
+			MonoClassField* field = (MonoClassField*) args [0]->inst_p1;
```

Comment:
This looks a little brittle, it depends on  OP_VMOVE never having its inst_p1 set otherwise.

Reply:
I guess I could also use `backend`; the comments there do not indicate that the field is used with OP_VMOVE. Another way would be to do what you suggested earlier - emit a new opcode that would carry it in `inst_p1` and then decompose it away. I had some issues getting that approach to work, so I tried this. It seems to work, although I cannot honestly verify that its `inst_p1` is always untouched. Wouldn't that also be the case for the new opcode?

Reply:
So now a `OP_LDTOKEN_FIELD` is emitted, which is equivalent in every way to a `OP_VMOVE`, except its opcode. This should protect the `MonoInst` contents until it is used by `CreateSpan` intrinsic. The decomposition of `OP_LDTOKEN_FIELD` is then only rewriting the opcode to `OP_VMOVE`.